packages/nextjs/components/ActivitiesItem.tsx

```diff
@@ -0,0 +1,225 @@
+import { CheckCircleIcon, DocumentDuplicateIcon } from "@heroicons/react/24/outline";
+import { getBlockExplorerAddressLink, getTargetNetwork } from "~~/utils/scaffold-eth";
+
+/**
+ * Displays an address (or ENS) with a Blockie image and option to copy address.
+ */
+
+const Address = ({ address }: { address: string }) => {
+  const [addressCopied, setAddressCopied] = useState(false);
+  const displayAddress = address?.slice(0, 5) + "..." + address?.slice(-4);
+  const blockExplorerAddressLink = getBlockExplorerAddressLink(getTargetNetwork(), address);
+
+  return (
+    <div className="flex">
+      {getTargetNetwork().id === hardhat.id ? (
+        <span className={`ml-1.5 text-sm font-normal`}>
+          <Link href={blockExplorerAddressLink}>{displayAddress}</Link>
+        </span>
+      ) : (
+        <a
+          className={`ml-1.5 text-sm font-normal`}
+          target="_blank"
+          href={blockExplorerAddressLink}
+          rel="noopener noreferrer"
+        >
+          {displayAddress}
+        </a>
+      )}
+      {addressCopied ? (
+        <CheckCircleIcon
+          className="ml-1.5 text-xl font-normal text-sky-600 h-5 w-5 cursor-pointer"
+          aria-hidden="true"
+        />
+      ) : (
+        <CopyToClipboard
+          text={address}
+          onCopy={() => {
+            setAddressCopied(true);
+            setTimeout(() => {
+              setAddressCopied(false);
+            }, 800);
+          }}
+        >
+          <DocumentDuplicateIcon
+            className="ml-1.5 text-xl font-normal text-sky-600 h-5 w-5 cursor-pointer"
+            aria-hidden="true"
+          />
+        </CopyToClipboard>
+      )}
+    </div>
+  );
+};
+
+export type TActivityItemProps = {
+  address: string;
+  amount: number;
+  landedOn: string;
+};
+
+export const ActivitiesItems = ({ address, amount, landedOn }: TActivityItemProps) => {
+  const [ensAvatar, setEnsAvatar] = useState<string | null>();
+
+  const { data: fetchedEnsAvatar } = useEnsAvatar({
+    address,
+    enabled: isAddress(address ?? ""),
+    chainId: 1,
+    cacheTime: 30_000,
+  });
+
+  const blockieSizeMap = {
+    xs: 6,
+    sm: 7,
+    base: 8,
+    lg: 9,
+    xl: 10,
+    "2xl": 12,
+    "3xl": 15,
+  };
+
+  const size = "3xl";
+
+  useEffect(() => {
+    setEnsAvatar(fetchedEnsAvatar);
+  }, [fetchedEnsAvatar]);
+
+  // Skeleton UI
+  if (!address) {
+    return (
+      <div className="animate-pulse flex space-x-4">
+        <div className="rounded-md bg-slate-300 h-6 w-6"></div>
+        <div className="flex items-center space-y-6">
+          <div className="h-2 w-28 bg-slate-300 rounded"></div>
+        </div>
+      </div>
+    );
+  }
+
+  if (!ethers.utils.isAddress(address)) {
+    return <span className="text-error">Wrong address</span>;
+  }
+
+  return (
+    <>
+      <div className="flex justify-between space-x-3 px-1 my-2 mb-5">
+        <div className="  w-1/6 flex-shrink-0">
+          {ensAvatar ? (
```

Comment:
why not use `Address` here? (and remove most of the logic in the component)

Reply:
Because of the Ui, I had to create a custom component to fit the UI  https://www.figma.com/file/D0YZXzXhNtaP8j0lcq2JZR/Dice?type=design&node-id=1%3A2&mode=design&t=9nWbwZgT95vcsicq-1